substrate/network-libp2p/src/network_state.rs

```diff
@@ -593,16 +632,23 @@ impl NetworkState {
 		}
 	}
 
+	/// Get the info on a peer, if there's an active connection.
+	pub fn peer_info(&self, who: PeerId) -> Option<PeerInfo> {
```

Comment:
I feel like this should be merged with `session_info`, but I haven't looked at how the higher-level layers use `session_info()`.